packages/grid/x-data-grid/src/components/containers/GridRootStyles.ts

```diff
@@ -115,6 +115,7 @@ export const GridRootStyles = styled('div', {
     outline: 'none',
     height: '100%',
     display: 'flex',
+    overflow: 'hidden',
```

Comment:
The solution works for displays `grid` and `flex` but for `table` it still [causes](https://codesandbox.io/s/datagrid-6-0-0-in-grid-forked-qgwo7j?file=/demo.tsx) the same issue. I played around and `display: grid` seems to solve all the cases, do you see any caveat in making root element's display property `grid` 🤔 
```suggestion
    display: 'grid',
```